rules-reviewed/eap7/eap72/maven-javax-to-jakarta.rhamt.xml

```diff
@@ -0,0 +1,258 @@
+<?xml version="1.0"?>
+<ruleset xmlns="http://windup.jboss.org/schema/jboss-ruleset" id="maven-javax-to-jakarta"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <metadata>
+        <description>
+            This ruleset provides analysis of Maven built applications that depend on Java EE artifacts, which were
+            replaced by Jakarta EE artifacts, starting with JBoss EAP 7.3.
+        </description>
+        <dependencies>
+            <addon id="org.jboss.windup.rules,windup-rules-javaee,3.0.0.Final" />
+            <addon id="org.jboss.windup.rules,windup-rules-java,3.0.0.Final" />
+        </dependencies>
+        <sourceTechnology id="eap" versionRange="(,7.2]" />
+        <targetTechnology id="eap" versionRange="[7.3,)" />        
+    </metadata>
+    <rules>
+        <!--https://issues.jboss.org/browse/WINDUPRULE-396 -->
+        <!-- com.sun.mail:javax.mail -> com.sun.mail:jakarta.mail -->
+        <rule id="maven-javax-to-jakarta-00001">
+            <when>
+                <project>
+                    <artifact groupId="com.sun.mail" artifactId="javax.mail" />
+                </project>
+            </when>
+            <perform>
+                <hint title="Move to Jakarta EE Maven Artifacts - com.sun.mail:javax.mail" effort="1" category-id="potential">
+                    <message>If you migrate your application to JBoss EAP 7.3.x, and want to ensure its Maven building, running or testing works as expected, use instead the Jakarta EE dependency with groupId `com.sun.mail`, and artifactId `jakarta.mail`.</message>
+                    <tag>JakartaEE</tag>
```

Comment:
Having also:

-  a [`<link>`](https://access.redhat.com/documentation/en-us/red_hat_application_migration_toolkit/4.2/html-single/rules_development_guide/index#link-syntax) tag to documentation would be better (even if high level doc about new artifacts and not specific for this artifact)
- a `<quickfix>` tag fits very well to cover the "replace this with that" use case so that the user, using the Eclipse Plugin, will just have to right click on the issue to have it fixed. It could be something like:
```Xml
<quickfix name="artifactId-mail" type="REPLACE">
    <replacement>jakarta.mail</replacement>
    <search>javax.mail</search>
</quickfix>
```

Reply:
About the link, I had that at first but the problem is that the documentation is yet to be done, and honestly wouldn't provide much more info than the message here. If I had a temp link tag  what do you suggest for its title and href attributes?
Wrt the quickfix can it be used in the cases that both groupId and artifactId changed?

Reply:
**Link**
If the documentation is not available yet, i think it would be fine to wait to merge this PR until we have the official link.
For a `<link>` tag, it's fine to have:

- title = "{guide title} - {paragraph title}" (e.g. [`JBoss EAP 7 - JAX-WS Web Services Endpoints`](https://github.com/windup/windup-rulesets/blob/master/rules-reviewed/eap6/java-ee/xml-webservices.windup.xml#L148))
- href = just the full link to the documentation

**Quickfix**
Yep, in that case you have to provide 2 different `<quickfix>` tags: 1 for groupId and 1 for artifactId

Reply:
Force pushed changes:
* added quickfix tags
* added link tags, with href attrs to be fixed, once eap 7.3 migration guide content is available